core/src/write.jl

```diff
@@ -328,14 +329,31 @@ end
 function write_arrow(
     path::AbstractString,
     table::NamedTuple,
-    compress::Union{ZstdCompressor, Nothing},
+    compress::Union{ZstdCompressor, Nothing};
+    init::Bool,
 )::Nothing
+    # At the start of the simulation, write an empty table to ensure we have permissions
+    # and fail early.
+    # At the end of the simulation, write all non-empty tables, and remove existing empty ones.
+    if !init && isempty(table.time)
```

Comment:
The order of simulation and the inline documentation are different from the code order. This confused me for a bit. Partially because the init keyword is only used inversely, to indicate the end of the simulation. Maybe rename the keyword to `remove_empty_table`?